strawberryfields/io.py

```diff
@@ -70,35 +71,15 @@ def to_blackbird(prog, version="1.0"):
 
             if cmd.op.p:
                 # argument is quadrature phase
-                op["kwargs"]["phi"] = cmd.op.p[0].x
+                op["kwargs"]["phi"] = cmd.op.p[0]
 
         else:
-            if cmd.op.p is not None:
-                for a in cmd.op.p:
-                    # check if reg ref transform
-                    if isinstance(a.x, RegRefTransform):
-                        # if a.x.func_str is not None:
-                            # TODO: will not satisfy all use cases
-                            # as the RegRefTransform string cannot be checked
-                            # to determine if it is a valid function for serialization!
-                            #
-                            # Possible solutions:
-                            #
-                            #   * Use SymPy to represent functions, as
-                            #     SymPy provides methods for converting to
-                            #     Python functions as well as serialization
-                            #
-                            #   * Don't allow classical processing of measurements
-                            #     on remote backends
-                            #
-                            # op["args"].append(a.x.func_str)
-                        # else:
-                        raise ValueError(
-                            "The RegRefTransform in operation {} "
-                            "is not supported by Blackbird.".format(cmd.op)
-                        )
-                    # else:
-                    op["args"].append(a.x)
+            for a in cmd.op.p:
+                if par_is_symbolic(a):
+                    # SymPy object, convert to string
+                    a = str(a)
+                op["args"].append(a)
```

Comment:
Nice! :+1:

Reply:
What should `op["args"]` contain?

The operation parameters in SF can now be (1) symbolic (incl. arrays with symbols in them), (2) numeric arrays, or (3) python numeric scalars. The code here converts case (1) to strings but stores other types as is, is that correct?

Reply:
I believe it should be converting case (1) to a 'templated' blackbird variable, i.e., for `Symbol(s) -> "{s}"`

Reply:
This isn't happening at the moment. Free parameters are replaced with their name, measured parameters by "q[k].par".

Reply:
So free params should be converted to "{parameter_name}" and measured parameters to "qx", is that correct? Do all the free params of a program have to be listed somewhere?

Reply:
Yep that's correct, and no they don't need to be listed

Reply:
See here: https://quantum-blackbird.readthedocs.io/en/latest/syntax.html#templates

Reply:
I just pushed the first half of compatibility with Blackbird template parameters (which correspond to FreeParameters) and measured parameters. I left a FIXME comment in `io.py` where the conversion of the symbolic expressions should take place, and two for-now skipped tests to test it.